src/connection.rs

```diff
@@ -180,7 +180,6 @@ pub enum ConnectionState {
 impl From<i64> for ConnectionState {
     fn from(state: i64) -> Self {
         match state {
-            0 => ConnectionState::Unknown,
```

Comment:
sounds good, it may be worth adding the extra states to the match statement aswell